command/sub8_missions/missions/collect_classify.py

```diff
@@ -0,0 +1,395 @@
+    def vision_response_to_pose(res):
+        '''
+        Returns a tuple (position, orientation) representing the pose
+        of a VisionRequest service response
+        '''
+        return (rosmsg_to_numpy(res.pose.pose.position), rosmsg_to_numpy(res.pose.pose.orientation))
+
+    @util.cancellableInlineCallbacks
+    def go_to_pinger(self):
+        '''
+        Move towards pinger at center of tower. Incrementally moves in direction
+        of pinger heading if available. Once position estimate is available, move
+        a few meters backwards from pinger.
+        '''
+        yield self.sub.passive_sonar.set_frequency(self.FREQ)
+        yield self.sub.passive_sonar.reset()
+        self.pinger_position = None
+        torward_move = None
+
+        @util.cancellableInlineCallbacks
+        def get_pinger_pose():
+            while self.pinger_position is None:
+                res = yield self.sub.passive_sonar.estimate_position()
+                if res.num_headings >= self.MIN_OBSERVATIONS:
+                    assert res.header.frame_id in ('/map', 'map')
+                    self.pinger_position = np.array([res.x, res.y, res.z])
+                yield self.sub.nh.sleep(0.5)
+
+        while self.pinger_position is None:
+            heading = yield self.sub.passive_sonar.get_heading()
+            if heading.success:
+                hydrophones_to_baselink = yield self.sub._tf_listener.get_transform('/map', heading.header.frame_id,
+                                                                                    heading.header.stamp)
+                self.pinger_heading = hydrophones_to_baselink.transform_vector(np.array([heading.x, heading.y,
+                                                                                         heading.z]))
+                self.pinger_heading[2] = 0.0
+                self.pinger_heading = self.pinger_heading / np.linalg.norm(self.pinger_heading)
+                goal = self.sub.pose.position + self.pinger_heading
+
+                torward_move = self.sub.move.look_at(goal).forward(1).go(blind=self.BLIND)
+                if self.pinger_position is None:
+                    yield torward_move
+                torward_move = None
+            else:
+                yield self.sub.nh.sleep(0.5)
+
+        print 'MOVING TO PONGO', self.pinger_position
+        yield self.sub.move.look_at(self.pinger_position).set_position(
+            self.pinger_position).backward(3.0).go(blind=self.BLIND)
+
+    @util.cancellableInlineCallbacks
+    def surface(self):
+        '''
+        Surface sub at current position, wait a few seconds,
+        then submerge again.
+        '''
+        yield self.sub.move.depth(0).go()
+        yield self.sub.nh.sleep(2.0).go()
+        depth = self.height_to_depth(CollectClassifyMission.OVAL_SEARCH_HEIGHT)
+        yield self.sub.move.depth(depth).go()
+
+    @util.cancellableInlineCallbacks
+    def circle(self, origin, radius, num_moves=10, proportion=1.0):
```

Comment:
I think they best place for this could be in the `_PoseProxy` class in the sub_singleton file. There are some "advanced moves" in there that this could fall under.